back/index.js

```diff
@@ -1,14 +1,36 @@
 const express = require('express');
 const mongoose = require('mongoose');
+const bodyParser = require('body-parser');
 
 const Product = require('./models/product');
 
-const App = express();
-const connect = mongoose.connect('mongodb://localhost:27017/express-react-shop');
+const app = express();
+const port = 5001;
 
-App.get('/products', async (req, res) => {
-  const products = await Product.find();
-  res.json(products);
-});
+app.use(bodyParser.urlencoded({
+  extended: true
+}));
+app.use(bodyParser.json());
 
-App.listen(3000, () => console.log('The server is started'));
+mongoose.connect(
+  'mongodb://localhost:27017/express-react-shop',
+  { useNewUrlParser: true }
+);
+
+app.route('/products')
+  .get(async (req, res) => {
+    const products = await Product.find();
+    res.json(products);
+  })
+  .post(async (req, res) => {
+    const product = await Product.create(req.body.product);
```

Comment:
Let's add some data validation. We need this to be able to properly handle server errors on the client side. Create an express middleware that runs before the `post` route. This middleware should validate the presence of `title, description, price, images` in the request body and respond with all validation errors if there are any.